Intermittent build failures due to timing issues in cruise.statemachine.test.DoActivityTest

My effort to commit little changes to the server is unsuccessful. The following is an excerpt from the log. Hope this can  help fix the issue. On my machine, nothing fails but building via travis is always failing.    

```
[junit] Running cruise.statemachine.test.DoActivityTest
[junit] Tests run: 8, Failures: 0, Errors: 1, Time elapsed: 36.854 sec
```

BUILD FAILED
/home/travis/build/umple/umple/build/build.testbed.xml:144: The following error occurred while executing this line:
/home/travis/build/umple/umple/build/build.testbed.xml:158: Tests failed

See also #591 that may be related or may have a suggested solution

--- a/testbed/src/TestHarnessStateMachineJava.ump
+++ b/testbed/src/TestHarnessStateMachineJava.ump
@@ -298,6 +298,7 @@ class CourseWMultiDo {
           addLog("DoActivity in nested state Thread1");
         }
         do {
+          Thread.sleep(500);
           addLog("DoActivity in nested state Thread2");
         }
       }  

--- a/testbed/test/cruise/statemachine/test/DoActivityTest.java
+++ b/testbed/test/cruise/statemachine/test/DoActivityTest.java
@@ -19,9 +19,7 @@ public class DoActivityTest
     Assert.assertEquals("Do Activity On Closed",course.getLog(2));
   }
   
-  @Test
-  @Ignore
-  // Ignored because was nondeterministic and breaking build when build was running under heavily loaded machine FIX FIX TO DO    
+  @Test   
   public void stopProcessingIfYouLeaveTheState() throws InterruptedException
   {
     CourseC course = new CourseC();

